### ISSUE TITLE
NullPointerException in DefaultConverterProvider when it tries to upcast a null reference

### ISSUE THREAD

**Issue by frlzk** [jOOQ/jOOQ#12089]
### Expected behavior
Not throw NullPointerException , return null ;
### Actual behavior
org.jooq.impl.DefaultConverterProvider$2.from(DefaultConverterProvider.java:155) throw NullPointerException
### Steps to reproduce the problem

java class is :
```java
class MyBean{
	private java.sql.Date birthday;

	public Date getBirthday() {
		return birthday;
	}

	public void setBirthday(Date birthday) {
		this.birthday = birthday;
	}
}
``` 

When birthday is null from database then throw exception:
```
org.jooq.exception.MappingException: An error ocurred when mapping record to class MyBean
	at org.jooq.impl.DefaultRecordMapper$MutablePOJOMapper.map(DefaultRecordMapper.java:874)
	at org.jooq.impl.DefaultRecordMapper.map(DefaultRecordMapper.java:548)
	at org.jooq.impl.ResultImpl.into(ResultImpl.java:1280)
	at org.jooq.impl.AbstractResultQuery.fetchInto(AbstractResultQuery.java:1522)
	at org.jooq.impl.SelectImpl.fetchInto(SelectImpl.java:3936)
	......
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.NullPointerException: null
	at org.jooq.impl.DefaultConverterProvider$2.from(DefaultConverterProvider.java:155)
	at org.jooq.impl.AbstractRecord.get(AbstractRecord.java:284)
	at org.jooq.impl.DefaultRecordMapper$MutablePOJOMapper.map(DefaultRecordMapper.java:901)
	at org.jooq.impl.DefaultRecordMapper$MutablePOJOMapper.map(DefaultRecordMapper.java:828)
``` 

### Versions

- jOOQ: 3.14.11
- Java: 1.8
- Database (include vendor): PostgreSql 11
- OS: Ubuntu
- JDBC Driver (include name if inofficial driver): postgresql-42.2.5.jar

**Comment by lukaseder** [comment]
Thanks a lot for your report. The failing lines in question are (in jOOQ 3.14):

https://github.com/jOOQ/jOOQ/blob/9e916d8/jOOQ/src/main/java/org/jooq/impl/DefaultConverterProvider.java#L151-L156

The issue doesn't persist in jOOQ 3.15, where a refactoring has been made already, to implement `null` safe conversion using `Converter.ofNullable()`: 

https://github.com/jOOQ/jOOQ/blob/b0085fd/jOOQ/src/main/java/org/jooq/impl/DefaultConverterProvider.java#L137-L147

**Comment by lukaseder** [comment]
This is a reproducer that fails for H2 on 3.14 and runs on 3.15:

```java
assertNull(create().fetchOne("select null").get(0, Object[].class));
```

**Comment by lukaseder** [comment]
Fixed for 3.14.12. 3.15.0 is not affected. Thanks again for your report!

**Comment by lukaseder** [comment]
3.14.12 has been released including this fix
